Sources/Account/Account+TransactionOperations.swift

```diff
@@ -281,5 +281,99 @@ extension Account {
+                switch $0 {
+                case .success(let feeStrategy):
+
+                    let (unspentTxOuts, ledgerBlockCount) =
+                        self.account.readSync {
+                            ($0.unspentTxOuts(tokenId: feeTokenId), $0.knowableBlockCount)
+                        }
+
+                    if ledgerBlockCount > presignedInput.tombstoneBlockIndex {
+                        serialQueue.async {
+                            completion(.failure(.invalidInput("Presigned Input Expired.")))
+                        }
+                        return
+                    }
+
+                    logger.info(
+                        "Preparing pre-signed input transaction with fee level...feeLevel: " +
+                            "\(feeLevel), unspentTxOutValues: " +
+                            "\(redacting: unspentTxOuts.map { $0.value })",
+                        logFunction: false)
+
+                    switch self.txOutSelector
+                        .selectTransactionInput(
+                            amount: amountToSend,
+                            feeStrategy: feeStrategy,
+                            fromTxOuts: unspentTxOuts)
+                        .mapError({ error -> TransactionPreparationError in
+                            switch error {
+                            case .insufficientTxOuts:
+                                return .insufficientBalance()
+                            case .defragmentationRequired:
+                                return .defragmentationRequired()
+                            }
+                        })
+                    {
+                    case .success(let (inputs: inputs, fee: fee)):
+                        metaFetcher.blockVersion {
+                            switch $0 {
+                            case .success(let blockVersion):
+                                logger.info(
+                                    "Transaction with presigned input prepared with fee level. " +
+                                        "fee: \(redacting: fee)",
+                                    logFunction: false)
+
+                                let tombstoneBlockIndex =
+                                    min(ledgerBlockCount + 50, presignedInput.tombstoneBlockIndex)
+
+                                self.transactionPreparer.preparePresignedInputTransaction(
+                                    presignedInput: presignedInput,
+                                    inputs: inputs,
+                                    memoType: memoType,
+                                    amount: amountToSend,
+                                    fee: Amount(fee, in: presignedInput.feeTokenId),
+                                    tombstoneBlockIndex: tombstoneBlockIndex,
+                                    blockVersion: blockVersion,
+                                    completion: completion)
+                            case .failure(let error):
```

Comment:
You could convert the fee & blockversion checks to `guards` to some of these nested .failure cases.

Reply:
will revisit nits after merge